packages/@react-stately/data/src/useListData.ts

```diff
@@ -326,11 +326,7 @@ function insert<T>(state: ListState<T>, index: number, ...values: T[]): ListStat
 
 function move<T>(state: ListState<T>, indices: number[], toIndex: number): ListState<T> {
   // Shift the target down by the number of items being moved from before the target
-  for (let index of indices) {
-    if (index < toIndex) {
-      toIndex--;
-    }
-  }
```

Comment:
Since the values in `indicies` are in ascending order, it is possible for `toIndex` to be decremented to the point where the final index in `indicies` is the same value as `toIndex`